docs/configuring-a-resource.md

```diff
@@ -22,126 +22,239 @@ cloud provider; sometimes it could simply be the name of resource
+```
 
-Hence, we end up the following external name configuration for `aws_s3_bucket`
-resource:
+To properly set external name for such a resource, we need to configure how to
+extract external name from this string (`GetExternalNameFn`) and how to build
+this id back (`GetIDFn`).
 
 ```go
-func Configure(p *config.Provider) {
-    p.AddResourceConfigurator("aws_s3_bucket", func(r *config.Resource) {
-        r.ExternalName = config.ExternalName{
-            SetIdentifierArgumentFn: func(base map[string]interface{}, name string) {
-                base["bucket"] = name
-            },
-            OmittedFields: []string{
-                "bucket",
-                "bucket_prefix",
-            },
-        }
-    })
+import (
+	"github.com/crossplane-contrib/terrajet/pkg/config"
+	...
+)
+
+func getNameFromFullyQualifiedID(tfstate map[string]interface{}) (string, error) {
+	id, ok := tfstate["id"]
+	if !ok {
+		return "", errors.Errorf(ErrFmtNoAttribute, "id")
+	}
+	idStr, ok := id.(string)
+	if !ok {
+		return "", errors.Errorf(ErrFmtUnexpectedType, "id")
+	}
+	words := strings.Split(idStr, "/")
+	return words[len(words)-1], nil
 }
+
+func getFullyQualifiedIDfunc(ctx context.Context, externalName string, parameters map[string]interface{}, providerConfig map[string]interface{}) (string, error)
+	subID, ok := providerConfig["subscription_id"]
+    if !ok {
+        return "", errors.Errorf(ErrFmtNoAttribute, "subscription_id")
+    }
+    subIDStr, ok := subID.(string)
+    if !ok {
+        return "", errors.Errorf(ErrFmtUnexpectedType, "subscription_id")
+    }
+    rg, ok := parameters["resource_group_name"]
+    if !ok {
+        return "", errors.Errorf(ErrFmtNoAttribute, "resource_group_name")
+    }
+    rgStr, ok := rg.(string)
+    if !ok {
+        return "", errors.Errorf(ErrFmtUnexpectedType, "resource_group_name")
+    }
+    
+	name, ok := parameters["name"]
+    if !ok {
+        return "", errors.Errorf(ErrFmtNoAttribute, "name")
+    }
+    nameStr, ok := rg.(string)
+    if !ok {
+        return "", errors.Errorf(ErrFmtUnexpectedType, "name")
+    }
+
+    return fmt.Sprintf("/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Sql/servers/%s", subIDStr, rgStr, nameStr), nil
+}
+
+...
+    p.AddResourceConfigurator("azurerm_sql_server", func(r *config.Resource) {
+        r.ExternalName = config.NameAsIdentifier
+        r.ExternalName.GetExternalNameFn = getNameFromFullyQualifiedID
+        r.ExternalName.GetIDFn = getFullyQualifiedIDfunc
+		...
+    }
 ```
 
-Please note, you can always check resource configurations of existing Providers
-as further examples under `config/<group>/config.go`.
+With this, we have covered most common scenarios for configuring external name.
+You can always check resource configurations of existing jet Providers as
+further examples under `config/<group>/config.go` in their repositories.
 
 ### Cross Resource Referencing
 
```

Comment:
In late-initialization behavior section, can we first describe the case where you'd need to customize and only then go into detail so that people can skip it in the beginning? In fact, we can have another high level section for configs like this one that should be done with data gathered from running the manual tests.